### ISSUE TITLE
Use `generates:` to force the execution of following `sources` with `--dry`

### ISSUE THREAD

**Issue by Ckarles** [go-task/task#493]
Currently, if task1 `generates` a set of files, and task2 `sources` these files, if task1 `sources` have changed the following shows with `--dry`:

```
task: [task1] <do-stuff>
task: Task "task2" is up to date
```
Then, during runtime, the condition gets re-computed and the second tasks executes.
It could be an issue because unexpected commands could be triggered even if not showing in the `--dry` output.

I believe an opposite behavior would be better, aka, by default, showing that task2 will be executed (as the `sources` of task2 intersect with the `generates` of task1), then, at runtime, stating that, because the sources are the same, there is no need to re-run task2.

tldr: it would be better when using `--dry` to show which task could run *at maxima*.

**Comment by SwitchedToGitlab** [comment]
@Ckarles any chance you can provide an example `Taskfile.yml` so I can understand this better? I'm not quite sure I follow you here.

**Comment by andreynering** [comment]
Hi @Ckarles,

I got what you meant: improve how `--dry` works by taking `generates` into consideration.

This makes sense, but I'd say it has a low priority to me as `--dry` is a secondary feature. I'll keep this issue open to see if more people find this improvement useful.

**Comment by trulede** [comment]
Closing stale issue.
